data/core/doc/highlighter.lua

```diff
@@ -10,15 +10,21 @@ local Highlighter = Object:extend()
 
 function Highlighter:new(doc)
   self.doc = doc
+  self.running = false
   self:reset()
+  self:start()
+end
 
-  -- init incremental syntax highlighting
+-- init incremental syntax highlighting
+function Highlighter:start()
+  if self.running then return end
+  self.running = true
   core.add_thread(function()
```

Comment:
Could we use the result of this call to keep track of the active coroutine? So we could check `core.threads[k]` to see if it's still active and avoid having to use `self.running`.

In the end it's basically be the same thing as we'd still have to save the return value of `core.add_thread`, so it should be fine the way it is now.

Reply:
ok, will leave it as it is now checking a boolean should be faster than a table